metaseq/optim/base_optimizer.py

```diff
@@ -108,7 +108,9 @@ def multiply_grads(self, c):
                     c = c.to(p.grad.device)
                 p.grad.data.mul_(c)
 
-    def clip_grad_norm(self, max_norm, norm_type="l2", aggregate_norm_fn=None):
+    def clip_grad_norm(
+        self, max_norm, norm_type="l2", aggregate_norm_fn=None, **kwargs
```

Comment:
Oh no - where did these kwargs come from?

Reply:
Lol I don't know now - I remember some test failed but now I can't reproduce 🤷

Reply:
Oh wait I found it - again in the unused tests, the `MetaseqAdam` class inherits `clip_grad_norm` from the `BaseOptimizer`, which doesn't accept a `skip_gradient_update_on_clip_norm` argument, but this is passed in from the `Trainer` object.  I'll change the base optimizer to accept skip_gradient_update_on_clip_norm

Reply:
We should probably promote some of these unused tests to being used again, lol. Opening https://github.com/facebookresearch/metaseq/issues/436 to track